build.c

```diff
@@ -246,13 +247,12 @@ printstatus(void)
 static int
 jobstart(struct job *j, struct edge *e)
 {
-	extern char **environ;
 	size_t i;
 	struct node *n;
 	struct string *rspfile, *content, *description;
 	int fd[2];
-	posix_spawn_file_actions_t actions;
-	char *argv[] = {"/bin/sh", "-c", NULL, NULL};
+	reproc_options options = { 0 };
+	const char *argv[] = {"/bin/sh", "-c", j->cmd->s, NULL};
```

Comment:
`j->cmd` doesn't get set until later on in this file, so I'm wondering how this is working.

Reply:
I never removed the correct assignment later in the file which luckily avoids the issue.